diffcore-rename.c

```diff
@@ -447,12 +447,16 @@ static int too_many_rename_candidates(int num_destinations, int num_sources,
 	 * growing larger than a "rename_limit" square matrix, ie:
 	 *
 	 *    num_destinations * num_sources > rename_limit * rename_limit
+	 *
+	 * We use st_mult() to check overflow conditions; in the
+	 * exceptional circumstance that size_t isn't large enough to hold
+	 * the multiplication, the system won't be able to allocate enough
+	 * memory for the matrix anyway.
 	 */
 	if (rename_limit <= 0)
 		rename_limit = 32767;
-	if ((num_destinations <= rename_limit || num_sources <= rename_limit) &&
-	    ((uint64_t)num_destinations * (uint64_t)num_sources
-	     <= (uint64_t)rename_limit * (uint64_t)rename_limit))
+	if (st_mult(num_destinations, num_sources)
+	    <= st_mult(rename_limit, rename_limit))
```

Comment:
In cases where the `rename_limit` is already greater than 65535, the `rename_limit * rename_limit` multiplication overflows and process halts. But I don't think an 'overflow error' is very helpful for the users in understanding what is the wrong with their configuration; i.e. `diff.renameLimit` documentation says nothing about a 'maximum allowed value'.

Reply:
@basarugur please note that the Git project does not do code reviews on GitHub, but on the mailing list instead. You can find the appropriate mail you want to reply to here: https://lore.kernel.org/git/pull.929.v3.git.git.1609272328.gitgitgadget@gmail.com/#r. Then, just follow the guidance at https://github.com/gitgitgadget/gitgitgadget/wiki/ReplyToThis how to send the reply.